src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -103,19 +106,22 @@ void sl_ble_event_handling_task(void)
     int32_t event_id;
 
     //! This semaphore is waiting for wifi module initialization.
-    rsi_semaphore_wait(&sl_rs_ble_init_sem, 0);
+    osSemaphoreAcquire(sl_rs_ble_init_sem, osWaitForever);
 
     // This function initialize BLE and start BLE advertisement.
     sl_ble_init();
 
     // Application event map
     while (1)
     {
-        //! This semaphore is waiting for next ble event task
-        rsi_semaphore_wait(&sl_ble_event_sem, 0);
-
         // checking for events list
         event_id = rsi_ble_app_get_event();
+        if (event_id == -1)
```

Comment:
This change here, I discussed with Shrinivas before, There is an internal ticket tracking this. Did we get a response from BLE platform on this?

This move of sempahore doesn't make sense

Reply:
Hi Junior,

This issue will be addressed in a separate PR , Shrinivas is working on it.

Reply:
Unresolving since the changes seems to still be present in this PR. Should it be fixed in another PR, then this changes should be removed from this one. As to not create any conflicts or reversing a potential fix.